### ISSUE TITLE
Adds error about variable being the same name as function in other module

### ISSUE THREAD

**Issue by lasseebert** [rrrene/credo#276]
### Environment

* Credo version (`mix credo -v`): `0.5.3`
* Erlang/Elixir version (`elixir -v`): 

```
Erlang/OTP 19 [erts-8.1] [source] [64-bit] [async-threads:10] [kernel-poll:false]

Elixir 1.3.4
```

* Operating system: Ubuntu 16.10

### What were you trying to do?

Running `mix credo` in a project with this file:

```elixir
defmodule Outer do
  @moduledoc false

  defmodule Inner do
    @moduledoc false

    def hello do
      nil
    end
  end

  def some_func do
    hello = "Hello"
    IO.puts(hello)
  end
end
```

### Expected outcome

No warnings

### Actual outcome

```
$ mix credo
Checking 182 source files (this might take a while) ...

  Warnings - please take a look                                                 
┃ 
┃ [W] ↗ Assigned variable `hello` has same name as a function in the same 
┃       module.
┃       lib/my_app/test.ex:13:5 (Outer.some_func)

Please report incorrect results: https://github.com/rrrene/credo/issues

Analysis took 18.9 seconds (0.3s to load, 18.6s running checks)
674 mods/funs, found 1 warning.

Showing priority issues: ↑ ↗ →  (use `--strict` to show all issues, `--help` for options).
```

Shouldn't `Outer.Inner` be treated as a different module than `Outer` here?

**Comment by rrrene** [comment]
Ouch. Yes, your are on point with your analysis.

José and I discussed deprecating these checks (NameRedecleration*) since they are doubling compiler gurantees (at least on Elixir `>= 1.4`) and are throwing questionable issues now and then.

Thx for reporting! :+1:

**Comment by lasseebert** [comment]
Ah ok, didn't know that about Elixir 1.4. I will just disable the check for now then. I guess I will upgrade to Elixir 1.4 soon :)

Thanks.

**Comment by rrrene** [comment]
The NameRedeclaration* checks try to warn about ambiguously named variables which are also defined as a function taking no parameters. In 1.4, you start getting warnings that all function invokations of this kind need to use `()` to indicate that it is a function call and not a variable, thus removing the original motivation for the Credo check.

So it's not exactly the same, but parts of it double the functionality provided by compiler. If that makes sense :D

**Comment by lasseebert** [comment]
It does make sense :)
